scripts/tests/chiptest/test_definition.py

```diff
@@ -208,15 +218,37 @@ def Run(self, runner, apps_register, paths: ApplicationPaths, pics_file: str):
 
         try:
             if self.target == TestTarget.ALL_CLUSTERS:
-                app_cmd = paths.all_clusters_app
+                target_app = paths.all_clusters_app
             elif self.target == TestTarget.TV:
-                app_cmd = paths.tv_app
+                target_app = paths.tv_app
             elif self.target == TestTarget.LOCK:
-                app_cmd = paths.lock_app
+                target_app = paths.lock_app
             else:
                 raise Exception("Unknown test target - "
                                 "don't know which application to run")
 
+            # The paths argument is a dataclass where each field contains a list of one application path
+            for field in paths.__dataclass_fields__:
+                path = getattr(paths, field)
+
+                # Do not add chip-tool to the register
+                if path == paths.chip_tool:
+                    continue
+
+                # For the app indicated by self.target, give it the 'default' key to add to the register
+                if path == target_app:
+                    key = 'default'
+                else:
+                    key = os.path.basename(path[0])
```

Comment:
I think I'll need to revisit generating a key for the non-default case. This was assuming I could use something like "chip-lock-app" as the key but on Linux, that key cannot be found so perhaps the app name can have some variations based on platform. I might go with something like using an index through the loop and calling str(index) to use as the key. Any other suggestions welcome.